package/MDAnalysis/core/groups.py

```diff
@@ -715,27 +730,41 @@ def center(self, weights, pbc=None, compound='group'):
         # enforce calculations in double precision:
         dtype = np.float64
 
-        if compound.lower() == 'group':
+        comp = compound.lower()
+        if comp == 'group':
             if pbc:
                 coords = atoms.pack_into_box(inplace=False)
             else:
                 coords = atoms.positions
+            # Weights don't make sense if we have less than 2 atoms:
+            if len(self) < 2:
```

Comment:
if we're going to put in optimisations for `len == 1` things we need tests too

Reply:
What about zero weights? This happens if dummy atoms are present.
I personally don't have a problem with defining the weighted center of a single atom as its position, *regardless* of a zero weight.

The same is *not* true when computing the weighted center of a group of atoms and the sum of the weights is zero.

Reply:
The behavior is now that the result vector is `nan` if `weights.sum() == 0`. This is now consistently enforced regardless of group size (also for groups with single atoms). I hope I have it all covered now.

Reply:
So it sounds like you settled on `np.nan` for zero sum weights? Seems right